x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/create_rules_schema.ts

```diff
@@ -59,6 +66,11 @@ export const createRulesSchema = Joi.object({
   timeline_id,
   timeline_title,
   meta,
+  ml_job_id: ml_job_id.when('type', {
+    is: 'machine_learning',
+    then: Joi.required(),
+    otherwise: Joi.forbidden(),
+  }),
```

Comment:
You can keep this name as abbreviated but I think this would be the first abbreviation we have in the REST API. It might be more descriptive to say "machine_learning_job_id" since type is "machine_learning" or just "job_id".

Since this will go through documentation, and people might be using the public API, I am just bringing it up now as once we add it we will never really be able to change it.

Reply:
Changed to machine_learning_job_id.